### PR TITLE
Feature/implement handler and service logic for saving trees

### PR DESCRIPTION
**Pull request by Mohammad96Assaf**
refs: 
#104 , #81

### CODE REVIEW

File: internal/entities/tree.go [hunk 1]
```diff
@@ -19,3 +19,12 @@ type Tree struct {
...
 }
+type TreeCreate struct {
+	TreeClusterID *int32
+	Readonly      bool
+	PlantingYear  int32
+	Species       string
+	Number        string
+	Latitude      float64
+	Longitude     float64
+}
```
Comment (choffmann) [inline internal/entities/tree.go:30]: Do we really need this?

File: internal/service/domain/tree/tree.go [hunk 2]
```diff
@@ -50,6 +53,42 @@ func handleError(err error) error {
...
+	}
+
+	fn := make([]entities.EntityFunc[entities.Tree], 0)
+	if treeCreate.TreeClusterID != nil {
+		treeClusterID, err := s.treeClusterRepo.GetByID(ctx, *treeCreate.TreeClusterID)
+		if err != nil {
+			return nil, handleError(err)
+		}
+		fn = append(fn, tree.WithTreeCluster(treeClusterID))
+	}
```
Comment (choffmann) [inline internal/service/domain/tree/tree.go:77]: If a tree is inserted into an existing tree cluster, the tree cluster coordinates must be updated

**Review by choffmann — CHANGES_REQUESTED** [review]

File: internal/entities/tree.go (hunk 1 above)
Comment (Mohammad96Assaf) [reply internal/entities/tree.go:30]: For `TreeCluster`, we have both `TreeClusterCreate` and `TreeClusterUpdate` structs for creating and updating, so I followed the same approach for Tree. What do you mean—are you suggesting we use the `TreeCreateRequest` directly in the handler without mapping it to another struct that has the same values?

Comment (choffmann) [reply internal/entities/tree.go:30]: Okay, that's a point... I had a different idea but the more I think about it the smarter it is to do it this way :+1:

**Review by choffmann — APPROVED** [review]

**Pull request merged**